File: ckml.py
from typing import List, Tuple

def parse_coords(text: str) -> List[Tuple[float, float]]:
    """Parse a KML coordinates string into ``(lat, lon)`` tuples."""
    coords = []
    for pair in text.strip().split():
        try:
            parts = pair.split(',')
            lon = float(parts[0])
            lat = float(parts[1])
        except Exception:
            coords.append((0.0, 0.0))
        else:
            coords.append((lat, lon))
    return coords

File: webui/src/ckml.js
export function parseCoords(text) {
  const result = [];
  for (const token of text.trim().split(/\s+/)) {
    const parts = token.split(',');
    if (parts.length < 2) {
      result.push([0, 0]);
      continue;
    }
    const lon = parseFloat(parts[0]);
    const lat = parseFloat(parts[1]);
    result.push([lat, lon]);
  }
  return result;
}
